### PR TITLE
Support self-host ClojureScript

### PR DESCRIPTION
**Pull request by arichiardi**
This patch introduces a tiny fix that makes the ClojureScript side compatible
with self-host environments like Lumo and Planck.

The tests can be launched, after installing `lumo`, with:

```shell
lumo -c $(lein with-profile +dev, classpath) -sfK -e '(require (quote pjstadig.run-all))' -e '(pjstadig.run-all/run)'
```

### CODE REVIEW

**Comment by arichiardi** [comment]
In normal ClojureScript the macros should still be required, I get 9 failures with both `lumo` and `lein cljsbuild test`.

**Comment by arichiardi** [comment]
Hold on this one, I need to confirm if it is working better. The `include-require` should not be necessary anymore but there are weird semantics with self-host I might have overlooked.

**Comment by arichiardi** [comment]
No, this one does not actually work, probably a better approach would be the one in https://github.com/pjstadig/humane-test-output/pull/8 for self-host.

**Comment by arichiardi** [comment]
So I pushed a change that basically adds the @ryfow feature but does not touch `humane-test-output.clj`.

We probably don't need to change both implementations, thoughts @pjstadig?

Another idea would be a friendly fork for cljs only.

**Comment by pjstadig** [comment]
This is @ryfow's change + a require cleanup? Does it still have the drawback of eliminating the error when `=` is used with only one argument?

**Comment by arichiardi** [comment]
I think this keeps the behavior on the Clojure side so I would guess it does preserve things.

Only the clojurescript side changes, let me test that .

**Comment by arichiardi** [comment]
Confirming *in ClojureScript* it does not error out in that case:

```clojure
(deftest a-test
  (is (= :oops-i-only-gave-one-argument)))

cljs.user=> (test/test-var #'a-test)
nil
```

**Comment by pjstadig** [comment]
A couple of thoughts:

1. I think assert-expr is the place to do the diff, not by modifying the event afterwards. Modifying the event afterwards requires inspecting the asserted form to determine if it has an '=' as the first element...this is essentially what assert-expr does, so why reimplement that?

2. Overriding assert-expr is messy, it sometimes interacts badly with IDEs and their own output formatting. humane-test-output really only works well with terminal output. If you are using an IDE, you should not be using humane-test-output, but instead relying on the IDEs output.

3. I don't want to lose my bonus feature. Any changes should preserve that.

All that said, I would like some things to be cleaned up with humane-test-output. For one I don't like the tests, I think an approach something like this would work better https://github.com/pjstadig/assert-diff/blob/master/test/pjstadig/assert_diff_test.clj

The `report-` function can certainly be cleaned up. There's more opportunity for pulling out the common (and not common) parts between the clj and cljs implementations.

I think there are probably some other clean ups that could occur with the clojurescript implementation, the clojure implementation, and how they share code/interact.

**Comment by pjstadig** [comment]
I'm going to move this conversation to #24.

File: src/pjstadig/util.cljc [hunk 1]
```diff
@@ -83,11 +85,32 @@
...
+        expected (if (seq diffs)
+                   (nth (second actual) 1)
+                   expected)]
+    (assoc event
+           :diffs diffs
+           :expected expected)))
+
+
 (defn define-fail-report []
-  #?(:clj (defmethod report :fail [& args]
```
Comment (arichiardi) [inline src/pjstadig/util.cljc:87]: @pjstadig would you then be ok if this PR could be brought in with the `#?(clj ` `defmethod` untouched and the `#?(cljs` part taking the new code path - no BONUS FEATURE?

**Pull request closed without merging**